airflow/ui/src/queries/useLogs.tsx

```diff
@@ -49,8 +50,47 @@ const parseLogs = ({ data }: ParseLogsProps) => {
     return { data, warning };
   }
 
-  // eslint-disable-next-line react/no-array-index-key
-  const parsedLines = lines.map((line, index) => <p key={index}>{line}</p>);
+  let startGroup = false;
+  let groupLines: Array<string> = [];
+  let groupName = "";
+
+  // TODO: Add support for nested groups
+  /* eslint-disable react/no-array-index-key */
+  const parsedLines = lines.map((line, index) => {
+    if (line.includes("::group::") && !startGroup) {
+      startGroup = true;
+      groupName = line.split("::group::")[1] as string;
```

Comment:
If you return here after starting a new group, the first line is not repeating the group header:
```suggestion
      groupName = line.split("::group::")[1] as string;

      return undefined;
```
...then it is like:
![image](https://github.com/user-attachments/assets/6328b2b1-2428-4d0a-ab6c-35e338e0afd6)